app/src/main/kotlin/cz/covid19cz/erouska/ui/main/ShortcutsManager.kt

```diff
@@ -0,0 +1,72 @@
+package cz.covid19cz.erouska.ui.main
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import cz.covid19cz.erouska.BuildConfig
+import cz.covid19cz.erouska.R
+import cz.covid19cz.erouska.service.CovidService
+
+class ShortcutsManager(private val context: Context) {
+
+    companion object {
+        object ShortcutsActions {
+            private val SCHEME = "erouska" + if (BuildConfig.FLAVOR == "dev") "-dev" else ""
+
+            val URL_PAUSE: Uri = Uri.parse("${SCHEME}://service/pause")
+            val URL_RESUME: Uri = Uri.parse("${SCHEME}://service/resume")
+        }
+    }
+
+    fun handlePossibleShortcut(intent: Intent) {
+        if (intent.action == Intent.ACTION_RUN) {
+            when (intent.data) {
+                ShortcutsActions.URL_PAUSE -> {
+                    context.startService(CovidService.pause(context))
+                }
+                ShortcutsActions.URL_RESUME -> {
+                    context.startService(CovidService.resume(context))
+                }
+            }
+        }
+    }
+
+    fun updateShortcuts(isRunning: Boolean) {
+        ShortcutManagerCompat.removeAllDynamicShortcuts(context)
+
+        val shortcuts = if (isRunning) {
+            val shortcut = ShortcutInfoCompat.Builder(context, "erouska-service")
+                .setShortLabel(context.getString(R.string.pause_app))
+                .setLongLabel(context.getString(R.string.pause_app))
+                .setIcon(IconCompat.createWithResource(context, R.drawable.ic_shortcut_pause))
+                .setIntent(
+                    Intent(
+                        Intent.ACTION_RUN,
+                        ShortcutsActions.URL_PAUSE
+                    ).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+                )
+                .build()
+
+            listOf(shortcut)
```

Comment:
NIT, you can return just the one shortcut from here and wrap it in that list below